test/e2e/storage/csi_mock/csi_volume_expansion.go

```diff
@@ -152,7 +153,7 @@ var _ = utils.SIGDescribe("CSI Mock volume expansion", func() {
 				}
 				if test.expectFailure {
 					err = testsuites.WaitForResizingCondition(ctx, pvc, m.cs, csiResizingConditionWait)
-					framework.ExpectError(err, "unexpected resizing condition on PVC")
+					gomega.Expect(err).To(gomega.HaveOccurred(), "unexpected resizing condition on PVC")
```

Comment:
Here and below there are still several wildcard `HaveOccurred` checks. While not wrong, most likely we can do better than that.

After a `WaitFor` checking for an error is probably wrong: the test should check that some condition is true consistently, not check for something else and expect a timeout.

Reply:
finally, refine all "HaveOccurred" checks.